### ISSUE TITLE
Creating consumer does not work with ActiveSupport gem activated

### ISSUE THREAD

**Issue by anykeyh** [nats-io/nats-pure.rb#67]
I come to a pretty nasty problem leading to not being able to use nats-pure with ActiveSupport gem.

Here goes the problem:

```ruby
        # nats/io/js.rb:344
        req = {
          stream_name: stream,
          config: config
        }

        result = api_request(req_subject, req.to_json, params)
```

It expects to call the method `NATS::JetStream::API::ConsumerConfig#to_json` defined in the same file.

But thanks (or no thanks...) to monkey patching from ActiveSupport, Hash#to_json is instead calling this method:

```ruby
  def to_json(options = nil)
      if options.is_a?(::JSON::State)
        # Called from JSON.{generate,dump}, forward it to JSON gem's to_json
        super(options)
      else
        # to_json is being invoked directly, use ActiveSupport's encoder
        ActiveSupport::JSON.encode(self, options)
      end
    end
```

Which itself is calling this block:

```ruby
  # in Hash object
  def as_json(options = nil) #:nodoc:
    if respond_to?(:to_hash)
      # struct respond_to to_hash
      to_hash.as_json(options)
    else
      instance_values.as_json(options)
    end
  end
```

This causes the to_json(*args) defined in the struct to be completely ignored and to never compact the configurations hash before sending to the API. This obviously doesn't play well with the nats server, causing this error:

```
invalid JSON (status_code=400, err_code=10025) (NATS::JetStream::Error::BadRequest)
```

The solution would be obviously not to use ActiveSupport, but since it's a pretty popular gem, I would recommend changing:

```ruby
        req = {
          stream_name: stream,
          config: config
        }

        result = api_request(req_subject, req.to_json, params)
```

to:

```ruby
        req = {
          stream_name: stream,
          config: config.to_h.compact
        }

        result = api_request(req_subject, req.to_json, params)
```
And remove the custom-defined `to_json` method in the structure, which is here only to compact anyway.

I could create a PR if you feel like `nats-pure` should work with active support core extensions.

**Comment by anykeyh** [comment]
For information in case someone get the same error, I was able to fix it temporarly by monkeypatching like this:

```ruby
# fix issue with consumer config.
class NATS::JetStream::API::ConsumerConfig
  # stupid activerecord.
  def as_json(*args)
    to_h.compact.as_json(*args)
  end
end
```

**Comment by wallyqs** [comment]
Thanks for the report @anykeyh, yes feel free to send a PR on what you feel could be a good change for this.

**Comment by wallyqs** [comment]
Fixed via #93
